src/components/learner-credit-management/BudgetAssignmentsTable.jsx

```diff
@@ -13,46 +13,138 @@ const BudgetAssignmentsTable = ({
   isLoading,
   tableData,
   fetchTableData,
-}) => (
-  <DataTable
-    isSortable
-    manualSortBy
-    isPaginated
-    manualPagination
-    isFilterable
-    manualFilters
-    isLoading={isLoading}
-    defaultColumnValues={{ Filter: TableTextFilter }}
-    FilterStatusComponent={FilterStatus}
-    columns={[
-      {
-        Header: 'Assignment details',
-        accessor: 'assignmentDetails',
-        Cell: AssignmentDetailsTableCell,
-        disableSortBy: true,
-      },
-      {
-        Header: 'Amount',
-        Cell: ({ row }) => `-${formatPrice(row.original.contentQuantity / 100, { maximumFractionDigits: 0 })}`,
-        disableFilters: true,
-      },
-    ]}
-    initialTableOptions={{
-      getRowId: row => row?.uuid?.toString(),
-    }}
-    initialState={{
-      pageSize: PAGE_SIZE,
-      pageIndex: DEFAULT_PAGE,
-      sortBy: [],
-      filters: [],
-    }}
-    fetchData={fetchTableData}
-    data={tableData.results}
-    itemCount={tableData.count}
-    pageCount={tableData.numPages}
-    EmptyTableComponent={CustomDataTableEmptyState}
-  />
-);
+}) => {
+  const [rowsSelected, setRowsSelected] = useState(false);
+
+  const CancelAction = ({ selectedFlatRows, ...rest }) => (
```

Comment:
Components should not be defined within another component, as described the ESLint rule:

```
Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “BudgetAssignmentsTable” and pass data as props.
```